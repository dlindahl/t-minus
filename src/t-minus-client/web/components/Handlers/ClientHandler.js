import { bindActionCreators } from 'redux';
import { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import ClientLayout from '../Layout/ClientLayout';
import Teleprompter from '../Teleprompter/Teleprompter';
import TimeDisplay from '../TimeDisplay/TimeDisplay';

function getState(state) {
  return {
    display: state.display,
    percentComplete: state.clock.percentComplete
  };
}

@connect(getState)
export default class ClientHandler extends Component {
  render() {
    return (
      <ClientLayout
        clockMode={this.props.clockMode}
        percentComplete={this.props.percentComplete}
      >
        <TimeDisplay {...this.props.display}/>
        <Teleprompter text={this.props.display.teleprompter}/>
      </ClientLayout>
    );
  }
}

ClientHandler.propTypes = {
  display: PropTypes.object
};
