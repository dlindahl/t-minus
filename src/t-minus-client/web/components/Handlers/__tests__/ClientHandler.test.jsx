import ClientHandler from '../ClientHandler'
import configureStore from 'redux-mock-store'
import emptyObj from 'empty/object'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

const STATES = {
  default: {
    clock: {
      percentComplete: 0
    },
    display: {
      teleprompter: 'TEXT'
    },
    meta: emptyObj
  }
}
const store = configureStore()

describe('ClientHandler', () => {
  it('renders the component', () => {
    const component = renderer.create(
      <Provider store={store(STATES.default)}>
        <ClientHandler/>
      </Provider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
