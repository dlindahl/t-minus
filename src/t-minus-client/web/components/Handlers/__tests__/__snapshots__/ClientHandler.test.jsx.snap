exports[`ClientHandler renders the component 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "height": "100vh",
    }
  }>
  <div>
    <form
      onSubmit={[Function]}
      style={
        Object {
          "alignItems": "center",
          "background": "var(--Clouds)",
          "display": "flex",
          "height": "100%",
          "paddingBottom": 5,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 5,
        }
      }>
      <label
        style={
          Object {
            "color": "var(--Concrete)",
            "fontFamily": "Roboto Mono, monospace",
            "textTransform": "uppercase",
          }
        }>
        Teleprompter:>
      </label>
      <span
        style={
          Object {
            "animation": "1s blink step-end infinite",
            "borderLeftColor": "var(--Concrete)",
            "borderLeftStyle": "solid",
            "borderLeftWidth": 2,
            "display": "inline-block",
            "height": "1.25em",
            "marginLeft": 5,
          }
        } />
      <input
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "background": "transparent",
            "border": "none",
            "color": "var(--Concrete)",
            "display": "inline-block",
            "flex": 1,
            "fontSize": 18,
            "fontWeight": 200,
            "outlineColor": "var(--Concrete)",
            "outlineWidth": "thin",
            "overflow": "hidden",
            "padding": 5,
          }
        } />
    </form>
  </div>
  <main
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
      }
    }>
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "var(--Clouds)",
          "color": "var(--MidnightBlue)",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "fontFamily": "Roboto Mono, monospace",
          "fontSize": "14vw",
          "justifyContent": "center",
          "width": "100%",
        }
      }>
      <div>
        <span
          data-test-id="prefix"
          style={
            Object {
              "opacity": 0,
              "pointerEvents": "none",
            }
          }>
          -
        </span>
        <span
          data-test-id="hours">
          --
        </span>
        :
        <span
          data-test-id="minutes">
          --
        </span>
        :
        <span
          data-test-id="seconds">
          --
        </span>
        <span
          data-test-id="microseconds"
          style={
            Object {
              "color": "var(--Silver)",
              "fontSize": "5vw",
            }
          }>
          .
          ----
        </span>
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "background": "var(--Clouds)",
          "color": "var(--MidnightBlue)",
          "display": "flex",
          "flex": 1,
          "fontFamily": "Roboto Mono, monospace",
          "fontSize": "8vw",
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingBottom": 10,
          "paddingLeft": "5%",
          "paddingRight": "5%",
          "paddingTop": 10,
          "textTransform": "uppercase",
          "transition": "flex-grow .5s ease-in-out",
        }
      }>
      <div
        data-test-id="prompterText">
        TEXT
      </div>
    </div>
  </main>
  <div
    style={
      Object {
        "background": "var(--Clouds)",
      }
    }>
    <nav
      style={
        Object {
          "height": 100,
        }
      }>
      <ul
        style={
          Object {
            "display": "flex",
            "height": "100%",
            "listStyleType": "none",
            "margin": 0,
            "padding": 0,
          }
        }>
        <li
          style={
            Object {
              "flex": 1,
              "textAlign": "center",
            }
          }>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "transparent",
                "border": 0,
                "color": "var(--Silver)",
                "cursor": "pointer",
                "fontSize": "2rem",
                "height": 50,
                "lineHeight": 1,
                "minWidth": 50,
                "padding": 0,
              }
            }>
            <svg
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "top",
                }
              }
              viewBox="0 0 40 40"
              width="1em">
              <g>
                <path
                  d="m20 33.4q4.8 0 8.2-3.5t3.4-8.3-3.4-8.2-8.2-3.4-8.2 3.4-3.4 8.2 3.4 8.3 8.2 3.5z m0-26.8q6.3 0 10.6 4.5t4.4 10.5-4.4 10.6-10.6 4.4-10.6-4.4-4.4-10.6 4.4-10.5 10.6-4.5z m0.9 6.8v8.7l6.6 3.9-1.2 2.1-7.9-4.7v-10h2.5z m-7.8-7.8l-7.6 6.4-2.1-2.5 7.6-6.4z m23.5 3.9l-2.1 2.6-7.6-6.5 2.1-2.5z" />
              </g>
            </svg>
            <br />
            <div
              style={
                Object {
                  "color": "var(--Silver)",
                  "fontFamily": "Roboto Mono, monospace",
                  "fontSize": ".8rem",
                  "lineHeight": 1.5,
                  "textTransform": "uppercase",
                }
              }>
              Timer
            </div>
          </button>
        </li>
        <li
          style={
            Object {
              "flex": 1,
              "textAlign": "center",
            }
          }>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "transparent",
                "border": 0,
                "color": "var(--Silver)",
                "cursor": "pointer",
                "fontSize": "2rem",
                "height": 50,
                "lineHeight": 1,
                "minWidth": 50,
                "padding": 0,
              }
            }>
            <svg
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "top",
                }
              }
              viewBox="0 0 40 40"
              width="1em">
              <g>
                <path
                  d="m20.9 11.6v8.8l7.5 4.4-1.3 2.2-8.7-5.4v-10h2.5z m-0.9 21.8c7.3 0 13.4-6.1 13.4-13.4s-6.1-13.4-13.4-13.4-13.4 6.1-13.4 13.4 6.1 13.4 13.4 13.4z m0-30c9.2 0 16.6 7.4 16.6 16.6s-7.4 16.6-16.6 16.6-16.6-7.4-16.6-16.6 7.4-16.6 16.6-16.6z" />
              </g>
            </svg>
            <br />
            <div
              style={
                Object {
                  "color": "var(--Silver)",
                  "fontFamily": "Roboto Mono, monospace",
                  "fontSize": ".8rem",
                  "lineHeight": 1.5,
                  "textTransform": "uppercase",
                }
              }>
              Stopwatch
            </div>
          </button>
        </li>
        <li
          style={
            Object {
              "flex": 1,
              "textAlign": "center",
            }
          }>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "transparent",
                "border": 0,
                "color": "var(--Silver)",
                "cursor": "pointer",
                "fontSize": "2rem",
                "height": 50,
                "lineHeight": 1,
                "minWidth": 50,
                "padding": 0,
              }
            }>
            <svg
              data-test-id="PlayArrow"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "top",
                }
              }
              viewBox="0 0 40 40"
              width="1em">
              <g>
                <path
                  d="m13.4 8.4l18.2 11.6-18.2 11.6v-23.2z" />
              </g>
            </svg>
            <br />
            <div
              style={
                Object {
                  "color": "var(--Silver)",
                  "fontFamily": "Roboto Mono, monospace",
                  "fontSize": ".8rem",
                  "lineHeight": 1.5,
                  "textTransform": "uppercase",
                }
              }>
              Start
            </div>
          </button>
        </li>
        <li
          style={
            Object {
              "flex": 1,
              "textAlign": "center",
            }
          }>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "transparent",
                "border": 0,
                "color": "var(--Silver)",
                "cursor": "pointer",
                "fontSize": "2rem",
                "height": 50,
                "lineHeight": 1,
                "minWidth": 50,
                "padding": 0,
              }
            }>
            <svg
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "top",
                }
              }
              viewBox="0 0 40 40"
              width="1em">
              <g>
                <path
                  d="m23.4 5h11.6v11.6h-3.4v-5.9l-16.3 16.3-2.3-2.3 16.3-16.3h-5.9v-3.4z m8.2 26.6v-11.6h3.4v11.6q0 1.4-1 2.4t-2.4 1h-23.2q-1.4 0-2.4-1t-1-2.4v-23.2q0-1.4 1-2.4t2.4-1h11.6v3.4h-11.6v23.2h23.2z" />
              </g>
            </svg>
            <br />
            <div
              style={
                Object {
                  "color": "var(--Silver)",
                  "fontFamily": "Roboto Mono, monospace",
                  "fontSize": ".8rem",
                  "lineHeight": 1.5,
                  "textTransform": "uppercase",
                }
              }>
              Present
            </div>
          </button>
        </li>
        <li
          style={
            Object {
              "flex": 1,
              "textAlign": "center",
            }
          }>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "transparent",
                "border": 0,
                "color": "var(--Silver)",
                "cursor": "pointer",
                "fontSize": "2rem",
                "height": 50,
                "lineHeight": 1,
                "minWidth": 50,
                "padding": 0,
              }
            }>
            <svg
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": undefined,
                  "verticalAlign": "top",
                }
              }
              viewBox="0 0 40 40"
              width="1em">
              <g>
                <path
                  d="m31.3 12.9c1.3 2 2.1 4.5 2.1 7.1 0 7.3-6.1 13.4-13.4 13.4v5l-6.6-6.8 6.6-6.6v5c5.5 0 10-4.4 10-10 0-1.7-0.5-3.3-1.2-4.7z m-11.3-2.9c-5.5 0-10 4.5-10 10 0 1.7 0.4 3.3 1.2 4.7l-2.4 2.4c-1.4-2-2.2-4.5-2.2-7.1 0-7.3 6.1-13.4 13.4-13.4v-5l6.6 6.8-6.6 6.6v-5z" />
              </g>
            </svg>
            <br />
            <div
              style={
                Object {
                  "color": "var(--Silver)",
                  "fontFamily": "Roboto Mono, monospace",
                  "fontSize": ".8rem",
                  "lineHeight": 1.5,
                  "textTransform": "uppercase",
                }
              }>
              Reset
            </div>
          </button>
        </li>
      </ul>
    </nav>
    <div
      style={
        Object {
          "background": "var(--Clouds)",
          "height": 50,
          "whiteSpace": "nowrap",
          "width": "0%",
        }
      } />
  </div>
</div>
`;
